src/number_theory/legendre_symbol/quadratic_char.lean

```diff
@@ -50,58 +53,50 @@ The interesting case is when the characteristic of `F` is odd.
 
 section quadratic_char
 
+open mul_char
+
 variables {F : Type*} [field F] [fintype F] [decidable_eq F]
 
 /-- Some basic API lemmas -/
-lemma quadratic_char_eq_zero_iff (a : F) : quadratic_char F a = 0 ↔ a = 0 :=
+lemma quadratic_char_eq_zero_iff (a : F) : quadratic_char' F a = 0 ↔ a = 0 :=
 begin
-  simp only [quadratic_char],
+  simp only [quadratic_char'],
   by_cases ha : a = 0,
   { simp only [ha, eq_self_iff_true, if_true], },
   { simp only [ha, if_false, iff_false],
     split_ifs; simp only [neg_eq_zero, one_ne_zero, not_false_iff], },
 end
 
 @[simp]
-lemma quadratic_char_zero : quadratic_char F 0 = 0 :=
-by simp only [quadratic_char, eq_self_iff_true, if_true, id.def]
+lemma quadratic_char_zero : quadratic_char' F 0 = 0 :=
+by simp only [quadratic_char', eq_self_iff_true, if_true, id.def]
 
 @[simp]
-lemma quadratic_char_one : quadratic_char F 1 = 1 :=
-by simp only [quadratic_char, one_ne_zero, is_square_one, if_true, if_false, id.def]
-
-/-- For nonzero `a : F`, `quadratic_char F a = 1 ↔ is_square a`. -/
-lemma quadratic_char_one_iff_is_square {a : F} (ha : a ≠ 0) :
-  quadratic_char F a = 1 ↔ is_square a :=
-by { simp only [quadratic_char, ha, (dec_trivial : (-1 : ℤ) ≠ 1), if_false, ite_eq_left_iff],
-     tauto, }
-
-/-- The quadratic character takes the value `1` on nonzero squares. -/
-lemma quadratic_char_sq_one' {a : F} (ha : a ≠ 0) : quadratic_char F (a ^ 2) = 1 :=
-by simp only [quadratic_char, ha, pow_eq_zero_iff, nat.succ_pos', is_square_sq, if_true, if_false]
+lemma quadratic_char_one : quadratic_char' F 1 = 1 :=
+by simp only [quadratic_char', one_ne_zero, is_square_one, if_true, if_false, id.def]
 
-/-- If `ring_char F = 2`, then `quadratic_char F` takes the value `1` on nonzero elements. -/
+/-- If `ring_char F = 2`, then `quadratic_char' F` takes the value `1` on nonzero elements. -/
 lemma quadratic_char_eq_one_of_char_two (hF : ring_char F = 2) {a : F} (ha : a ≠ 0) :
-  quadratic_char F a = 1 :=
+  quadratic_char' F a = 1 :=
 begin
-  simp only [quadratic_char, ha, if_false, ite_eq_left_iff],
+  simp only [quadratic_char', ha, if_false, ite_eq_left_iff],
   intro h,
   exfalso,
   exact h (finite_field.is_square_of_char_two hF a),
 end
 
-/-- If `ring_char F` is odd, then `quadratic_char F a` can be computed in
+/-- If `ring_char F` is odd, then `quadratic_char' F a` can be computed in
 terms of `a ^ (fintype.card F / 2)`. -/
-lemma quadratic_char_eq_pow_of_char_ne_two (hF : ring_char F ≠ 2) {a : F} (ha : a ≠ 0) :
-  quadratic_char F a = if a ^ (fintype.card F / 2) = 1 then 1 else -1 :=
+lemma quadratic_char_eq_pow_of_char_ne_two' (hF : ring_char F ≠ 2) {a : F} (ha : a ≠ 0) :
```

Comment:
Same.

Reply:
This already has a version for `quadratic_char`.